### ISSUE TITLE
No health regain out of combat with ress sickness on.

### ISSUE THREAD

**Issue by Phatcat** [elysium-project/server#2160]
No health regain out of combat. Stuck at 79%.

Rogue, level 37, has resurrection sickness.

![billede](https://user-images.githubusercontent.com/1907820/29486044-7356804c-84dd-11e7-9290-1f6a0d7b12f9.png)

As soon as ress sickness ran out and stats were set back to normal, out of combat regain kicked back in.

**Comment by Qynnix** [comment]
Well you get ALL your stats reduced by 75%, including the ones that affect health regen such as spirit so I'd say it's now a bug, your health should regen yes, but at a MUCH lower pace than normal.
Resurrection sickness basically takes you off the active part of the game so eat, trade, repair, travel and wait for it to fade.

**Comment by Phatcat** [comment]
As I stated, there was 0 regain, not 'an abnormally low regain'.

**Comment by Phatcat** [comment]
> Resurrection sickness basically takes you off the active part of the game so eat, trade, repair, travel and wait for it to fade.

You think I don't know what ress sickness is?

Why are you commenting as if I don't know what ress sickness is? even if I only had 10 spirit (which, a rogue level 37 has more than that simply as base stats) and you take away 75% of that that'd still leave 2 - 3 spirit.

**Comment by Cabro** [comment]
It's the same for warriors, ZERO HP regenerated during res sickness.
This was the same on other servers, and the reason is that the regen based on spirit for those classes was a bit off. The numbers need to be changed a bit so warriors and rogues can regen HP during sickness.

**Comment by GuybrushGit** [comment]
This happens because of the health regen formula:

> case CLASS_DRUID:   regen = (Spirit * 0.11 + 1);    break;
> case CLASS_HUNTER:  regen = (Spirit * 0.43 - 5.5);  break;
> case CLASS_MAGE:    regen = (Spirit * 0.11 + 1);    break;
> case CLASS_PALADIN: regen = (Spirit * 0.25);        break;
> case CLASS_PRIEST:  regen = (Spirit * 0.15 + 1.4);  break;
> case CLASS_ROGUE:   regen = (Spirit * 0.84 - 13);   break;
> case CLASS_SHAMAN:  regen = (Spirit * 0.28 - 3.6);  break;
> case CLASS_WARLOCK: regen = (Spirit * 0.12 + 1.5);  break;
> case CLASS_WARRIOR: regen = (Spirit * 1.26 - 22.6); break;

So if a warrior with rez sickness has 10 spirit his regen will be a negative value.
No idea where these numbers come from, did a quick search for vanilla regen rates and found nothing decent.

**Comment by Cabro** [comment]
Yes they should 100% give a positive number for rogues and warriors. Why? Because you could regen HP on retail with sickness. Just tweak the numbers a bit. Up to the devs to find a decent formula.

**Comment by GuybrushGit** [comment]
Best source for regen rates is this: https://wow.gamepedia.com/index.php?title=Health_regeneration&oldid=807024
But to apply those the base stats need to be reviewed, with those formulas  and our current base stats a level 2 warrior will regen to full in 3 ticks.

**Comment by Phatcat** [comment]
... Here are some better sources;

https://wow.gamepedia.com/Talk:Health_regeneration/Archive01
http://wowwiki.wikia.com/wiki/Talk:Health_regeneration

Is it wowpedia or wowwiki simply copy-pasing their info from the other? Oh well, totally unrelated.

anyways...

You should also regain health faster when sitting down;

http://wowwiki.wikia.com/wiki/Health_regeneration?oldid=222101
> Health Regeneration is increased while sitting.

https://wow.gamepedia.com/index.php?title=Health_regeneration&oldid=222101
> Health Regeneration is increased while sitting.

Jesus comeon, they even have the same oldid across gamepedia and wowwiki now? both 222101.... shameless.

Which is confirmed by these comments back from 2006;
http://wow.allakhazam.com/forum.html?forum=21&mid=1138070096228285115

... aaaaaaaaand, by blizzard themselves;

http://web.archive.org/web/20061115212026/http://www.worldofwarcraft.com/info/basics/firstfewlevels.html
> You regain health faster when sitting down. 


https://github.com/elysium-project/server/blob/development/src/game/Objects/Player.cpp#L5034
all these formulas honestly look like shit. Also who the heck came up with the bright idea of a negative constant? -.-

example; while they pre tbc on the wiki wasn't sure about the warrior, post tbc they figured it out for pre tbc;

http://wowwiki.wikia.com/wiki/Health_regeneration?oldid=222101
> Warrior: Spirit * 0.80 + ? 

http://wowwiki.wikia.com/wiki/Health_regeneration
> Warrior Pre-BC | Spirit * 0.80 + 6

As you can see, Spirit * 0.8 + 6 is a hell of a lot cleaner than... Spirit * 1.26 - 22.6 ... f' me, really... what horrible guesswork.

**Comment by GuybrushGit** [comment]
Thats essentialy the same that I posted. 
The formula looks solid but to be implemented the base attributes need to be reviewed. With those formulas low level characters will have insane regen speeds since the base spirit is probably too high.

**Comment by Phatcat** [comment]
> Thats essentialy the same that I posted.

Yeah... sure. Not like half of my comment and most of my links are about regaining while sitting down or anything, which you've not uttered a single word about. But good on you to outright repeat yourself about the base stats needing to be reviewed, but are you sure people got it? How about stating it a third time?

**Comment by GuybrushGit** [comment]
I ignored those comments because sitting regen is not relevant to this issue and is already implemented. Don't know why you keep bringing that up.

**Comment by Phatcat** [comment]
You're right, indeed so it does. I must have missed that looking over the functin the first time, but if my eyes ain't fooling me now it's increasing the rate by 50%, which going by the comments from that old allakhazam discussion (which first off says it's not affected by spirit at all which it is in the core) as well as your own provided link proves is way too much, which will only add to the problem with "low level characters will have insane regen speeds" - something you indeed seem to find relevant to this issue since you were the one to bring that up initially.
